package/kernel/linux/modules/lib.mk

```diff
@@ -98,6 +98,132 @@ endef
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=BLAKE2s hash function library
+  KCONFIG:=CONFIG_CRYPTO_LIB_BLAKE2S
+  FILES:= \
+	$(LINUX_DIR)/lib/crypto/libblake2s.ko \
+	$(LINUX_DIR)/lib/crypto/libblake2s-generic.ko
+endef
+
+define KernelPackage/lib-crypto-blake2s/x86/64
+  KCONFIG+=CONFIG_CRYPTO_BLAKE2S_X86
+  FILES+=$(LINUX_DIR)/arch/x86/crypto/blake2s-x86_64.ko
+endef
+
+$(eval $(call KernelPackage,lib-crypto-blake2s))
+
+
+define KernelPackage/lib-crypto-chacha20
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=ChaCha library interface
+  KCONFIG:=CONFIG_CRYPTO_LIB_CHACHA
+  FILES:=$(LINUX_DIR)/lib/crypto/libchacha.ko
+endef
+
+define KernelPackage/lib-crypto-chacha20/x86/64
+  KCONFIG+=CONFIG_CRYPTO_CHACHA20_X86_64
+  FILES+=$(LINUX_DIR)/arch/x86/crypto/chacha-x86_64.ko
+endef
+
+define KernelPackage/lib-crypto-chacha20/arm-neon
+  KCONFIG+=CONFIG_CRYPTO_CHACHA20_NEON
+  FILES+=$(LINUX_DIR)/arch/arm/crypto/chacha-neon.ko
+endef
+
+KernelPackage/lib-crypto-chacha20/imx6=$(KernelPackage/lib-crypto-chacha20/arm-neon)
+KernelPackage/lib-crypto-chacha20/ipq40xx=$(KernelPackage/lib-crypto-chacha20/arm-neon)
+KernelPackage/lib-crypto-chacha20/mvebu/cortexa9=$(KernelPackage/lib-crypto-chacha20/arm-neon)
+
+define KernelPackage/lib-crypto-chacha20/arm64-neon
+  KCONFIG+=CONFIG_CRYPTO_CHACHA20_NEON
+  FILES+=$(LINUX_DIR)/arch/arm64/crypto/chacha-neon.ko
+endef
+
+define KernelPackage/lib-crypto-chacha20/mips
+  KCONFIG+=CONFIG_CRYPTO_CHACHA_MIPS
+  FILES:=$(LINUX_DIR)/arch/mips/crypto/chacha-mips.ko
+endef
+
+KernelPackage/lib-crypto-chacha20/ath79=$(KernelPackage/lib-crypto-chacha20/mips)
+KernelPackage/lib-crypto-chacha20/ramips=$(KernelPackage/lib-crypto-chacha20/mips)
+
+$(eval $(call KernelPackage,lib-crypto-chacha20))
+
+
+define KernelPackage/lib-crypto-chacha20poly1305
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=ChaCha20-Poly1305 AEAD support (8-byte nonce library version)
+  KCONFIG:=CONFIG_CRYPTO_LIB_CHACHA20POLY1305
+  DEPENDS:=+kmod-lib-crypto-chacha20 +kmod-lib-crypto-poly1305
+  FILES:=$(LINUX_DIR)/lib/crypto/libchacha20poly1305.ko
+endef
+
+$(eval $(call KernelPackage,lib-crypto-chacha20poly1305))
+
+
+define KernelPackage/lib-crypto-curve25519
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=Curve25519 scalar multiplication library
+  KCONFIG:=CONFIG_CRYPTO_LIB_CURVE25519
+  FILES:= \
+	$(LINUX_DIR)/lib/crypto/libcurve25519.ko \
+	$(LINUX_DIR)/lib/crypto/libcurve25519-generic.ko
+endef
+
+define KernelPackage/lib-crypto-curve25519/x86/64
+  KCONFIG+=CONFIG_CRYPTO_CURVE25519_X86
+  FILES+=$(LINUX_DIR)/arch/x86/crypto/curve25519-x86_64.ko
+endef
+
+define KernelPackage/lib-crypto-curve25519/arm-neon
+  KCONFIG+=CONFIG_CRYPTO_CURVE25519_NEON
+  FILES+=$(LINUX_DIR)/arch/arm/crypto/curve25519-neon.ko
+endef
+
+KernelPackage/lib-crypto-curve25519/imx6=$(KernelPackage/lib-crypto-curve25519/arm-neon)
+KernelPackage/lib-crypto-curve25519/ipq40xx=$(KernelPackage/lib-crypto-curve25519/arm-neon)
+KernelPackage/lib-crypto-curve25519/mvebu/cortexa9=$(KernelPackage/lib-crypto-curve25519/arm-neon)
```

Comment:
Are there specific patforms that should request the accelerated x86_64 one, or will that happen automatically?

Reply:
Yeah, probably, I am just not signing up for adding the exhaustive list of platforms right now, so some of these definitions are currently unused (and untested)

Reply:
FTR I am testing my build against a MIPS mt7621 target. Looking to the community to fill in the blanks here.